File: calculadora.c
// Patricia Zaragoza Palma 

#include <stdio.h>

// Funciones para las operaciones
float sumar(float a, float b) {
    return a + b;
}

float restar(float a, float b) {
    return a - b;
}

float multiplicar(float a, float b) {
    return a * b;
}

float dividir(float a, float b) {
    if (b == 0) {
        printf("Error: División por cero no permitida.\n");
        return 0;
    }
    return a / b;
}

int main() {
    int opcion;
    float num1, num2, resultado;

    do {
        // Mostrar el menú
        printf("Calculadora\n");
        printf("1. Sumar\n");
        printf("2. Restar\n");
        printf("3. Multiplicar\n");
        printf("4. Dividir\n");
        printf("5. Salir\n");
        printf("Elige una opción (1-5): ");
        scanf("%d", &opcion);

        if (opcion >= 1 && opcion <= 4) {
            // Leer los números
            printf("Ingresa el primer número: ");
            scanf("%f", &num1);
            printf("Ingresa el segundo número: ");
            scanf("%f", &num2);
        }

        switch (opcion) {
            case 1:
                resultado = sumar(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 2:
                resultado = restar(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 3:
                resultado = multiplicar(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 4:
                resultado = dividir(num1, num2);
                printf("Resultado: %.2f\n", resultado);
                break;
            case 5:
                printf("Saliendo de la calculadora.\n");
                break;
            default:
                printf("Opción no válida. Por favor elige entre 1 y 5.\n");
        }

        printf("\n");
    } while (opcion != 5);

    return 0;
}

File: calculadora.py
# Autor: Patricia Zaragoza Palma
# Ingeniera en sistemas computacionales

def sumar(a, b):
    return a + b

def restar(a, b):
    return a - b

def multiplicar(a, b):
    return a * b

def dividir(a, b):
    if b == 0:
        return "Error: División por cero no permitida"
    return a / b

def calculadora():
    print("Calculadora")
    print("Operaciones disponibles:")
    print("1. Sumar")
    print("2. Restar")
    print("3. Multiplicar")
    print("4. Dividir")

    while True:
        opcion = input("Elige una opción (1/2/3/4) o 'q' para salir: ")

        if opcion.lower() == 'q':
            print("Saliendo de la calculadora.")
            break

        if opcion in ['1', '2', '3', '4']:
            try:
                num1 = float(input("Ingresa el primer número: "))
                num2 = float(input("Ingresa el segundo número: "))

                if opcion == '1':
                    print(f"Resultado: {sumar(num1, num2)}")
                elif opcion == '2':
                    print(f"Resultado: {restar(num1, num2)}")
                elif opcion == '3':
                    print(f"Resultado: {multiplicar(num1, num2)}")
                elif opcion == '4':
                    print(f"Resultado: {dividir(num1, num2)}")
            except ValueError:
                print("Error: Por favor ingresa números válidos.")
        else:
            print("Opción no válida. Por favor, elige una opción entre 1 y 4.")

if __name__ == "__main__":
    calculadora()
